lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py

```diff
@@ -130,27 +138,25 @@ def set_host_configuration_facts(self):
                     # Make sure option_type is defined some values do not have
                     # it defined and appear to be read only.
                     if 'option_type' in host_facts[option_key]:
-                        # We handle all supported types here so we can give
-                        # meaningfull errors.
+                        # We handle all supported types here so we can give meaningful errors.
                         option_type = host_facts[option_key]['option_type']
                         if self.is_boolean(option_value) and isinstance(option_type, vim.option.BoolOption):
                             option_value = self.is_truthy(option_value)
-                        elif isinstance(option_value, six.integer_types) and isinstance(option_type, vim.option.IntOption):
+                        elif (isinstance(option_value, integer_types) or self.is_integer(option_value))\
+                                and isinstance(option_type, vim.option.IntOption):
                             option_value = VmomiSupport.vmodlTypes['int'](option_value)
-                        elif isinstance(option_value, six.integer_types) and isinstance(option_type, vim.option.LongOption):
+                        elif (isinstance(option_value, integer_types) or self.is_integer(option_value, 'long'))\
+                                and isinstance(option_type, vim.option.LongOption):
                             option_value = VmomiSupport.vmodlTypes['long'](option_value)
                         elif isinstance(option_value, float) and isinstance(option_type, vim.option.FloatOption):
                             pass
-                        elif isinstance(option_value, six.string_types) and isinstance(option_type, (vim.option.StringOption, vim.option.ChoiceOption)):
+                        elif isinstance(option_value, string_types) and isinstance(option_type, (vim.option.StringOption, vim.option.ChoiceOption)):
                             pass
                         else:
-                            self.module.fail_json(
-                                msg="Provided value is of type %s.  Option %s expects:\n %s" %
-                                (type(option_value), option_key, str(option_type)))
+                            self.module.fail_json(msg="Provided value is of type %s."
+                                                      " Option %s expects: %s" % (type(option_value), option_key, type(option_type)))
```

Comment:
I converted option_type to a string due to the significant amount of useful information it provides (min, max, default, choices for choiceOptions)  If the formatting of the output the reason for changing it to a type call perhaps we should parse out the other values as well, and not just in this error but also in the try except when the module is called below.